.github/workflows/all-green.yml

```diff
@@ -14,15 +18,34 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       checks: read
-      contents: read
+      statuses: read
     steps:
-      - uses: wechuli/allcheckspassed@e22f45a4f25f4cf821d1273705ac233355400db1
+      - uses: DataDog/ensure-ci-success@f40e6ffd8e60280d478b9b92209aaa30d3d56895
         with:
-          delay: 10  # wait for this delay before starting
-          polling_interval: 1 # after a failure, how long do it wait before checking again
-          retries: 60  # how many retries before stopping 
-          # checks_exclude: ''  # comma separated list of checks to exclude, nothing right now
+          initial-delay-seconds: 10  # wait for this delay before starting
+          max-retries: 60  # how many retries before stopping 
+          ignored-name-patterns: |
+            consolidated-pipeline
```

Comment:
Ironically, this is an overall success status for the whole azure devops pipeline. So we could simply require that 😅

Reply:
One pain point would be flaky management : if at some point one of your job is flaky, and it takes more than few hours to fix, for PR merge, you need to : 

* remove the requirement for all gitlab jobs
* or harass repo admin to merge PRs

With this, you can temporary add the failing job in this list, and keep your green CI policy for all other jobs.

----

One another pitfall is new workflows. On Github PR engine, jobs are optional by default (what a shame), unless you add them in required list.
With that solution, it changes this logic : all jobs are required by default, **unless** you add it in the exclusion list.